rllib/core/testing/tf/bc_module.py

```diff
@@ -0,0 +1,121 @@
+        layers.append(tf.keras.layers.ReLU())
+        layers.append(tf.keras.layers.Dense(output_dim))
+
+        self.policy = tf.keras.Sequential(layers)
+        self._input_dim = input_dim
+        self._output_dim = output_dim
+
+    @override(RLModule)
+    def input_specs_exploration(self) -> ModelSpec:
+        return ModelSpec(self._default_inputs())
+
+    @override(RLModule)
+    def input_specs_inference(self) -> ModelSpec:
+        return ModelSpec(self._default_inputs())
+
+    @override(RLModule)
+    def input_specs_train(self) -> ModelSpec:
+        return ModelSpec(self._default_inputs())
+
+    @override(RLModule)
+    def output_specs_exploration(self) -> ModelSpec:
+        return ModelSpec(self._default_outputs())
+
+    @override(RLModule)
+    def output_specs_inference(self) -> ModelSpec:
+        return ModelSpec(self._default_outputs())
+
+    @override(RLModule)
+    def output_specs_train(self) -> ModelSpec:
+        return ModelSpec(self._default_outputs())
+
+    @override(RLModule)
+    def _forward_inference(self, batch: NestedDict) -> Mapping[str, Any]:
+        obs = batch[SampleBatch.OBS]
+        action_logits = self.policy(obs)
+        action_logits_inference = tf.argmax(action_logits, axis=-1)
+        action_dist = tfp.distributions.Deterministic(action_logits_inference)
+        return {"action_dist": action_dist}
+
+    @override(RLModule)
+    def _forward_exploration(self, batch: NestedDict) -> Mapping[str, Any]:
+        return self._forward_inference(batch)
+
+    @override(RLModule)
+    def _forward_train(self, batch: NestedDict) -> Mapping[str, Any]:
+        obs = batch[SampleBatch.OBS]
+        action_logits = self.policy(obs)
+        action_dist = tfp.distributions.Categorical(logits=action_logits)
+        return {"action_dist": action_dist}
+
+    @override(RLModule)
+    def get_state(self) -> Mapping[str, Any]:
+        return {"policy": self.policy.get_weights()}
+
+    @override(RLModule)
+    def set_state(self, state: Mapping[str, Any]) -> None:
+        self.policy.set_weights(state["policy"])
+
+    @override(TfRLModule)
+    def trainable_variables(self) -> NestedDict[tf.Tensor]:
+        return NestedDict({"policy": self.policy.trainable_variables})
+
+    @classmethod
+    @override(RLModule)
+    def from_model_config(
+        cls,
+        observation_space: "gym.Space",
+        action_space: "gym.Space",
+        model_config: Mapping[str, Any],
+    ) -> Union["RLModule", Mapping[str, Any]]:
+
+        config = {
+            "input_dim": observation_space.shape[0],
+            "hidden_dim": model_config["hidden_dim"],
+            "output_dim": action_space.n,
+        }
+
+        return cls(**config)
+
+    @classmethod
+    def from_env(cls, env: gym.Env):
+        """This is used for testing purposes."""
+        return cls.from_model_config(
+            env.observation_space,
+            env.action_space,
+            model_config={"hidden_dim": 32},
+        )
```

Comment:
remove? this API is not part of the RLModule API now. to make things simpler/cleaner.

Reply:
I can. Kourosh added this to the bc_module for torch so I added it myself. I can remove it in both places

Reply:
removed.